### ISSUE TITLE
`dinitctl replace`: a new command for replacing services

### ISSUE THREAD

**Issue by mobin-2008** [davmac314/dinit#221]
Hi.

Sometimes sysadmins need to change their system mode in runtime, for example form multi user mode into the single user mode (for maintenance purposes) or just need to _replace_ services with each other.

This new command will be used for:
1. Easily replacing some services which have conflicts between them: (such as `nginx` & `apache2`, `gdm` & `lxdm` and ...). For example: `dinitctl replace gdm sddm` instead of `dinitctl stop gdm; dinitctl start sddm`.

2. Changing the system mode, One of the purposes of this command is to replace `boot` service with something else. for example `dinitctl replace boot single` and system goes into the single user mode without rebooting or manually stopping `boot` (which I cannot did that).

Also it has an advantage over manually stopping & starting services:
- `x` has depends-on `y`. Also `g` has depends-on `y`. If we stop `x`, `y` will be stopped due to Dinit design. and on `g` starting process, `y` comes up again. I think it's unnecessary overhead. `dinitctl replace` will
  1. Create a _placeholder_ service which has same depends as `y`
  2. Stop `x` service
  3. Replace that _placeholder_ with actual `y` service

   I think it's good advantage to have something like this.

I think it's enough for introduction.
Regards

**Comment by davmac314** [comment]
Need to think about this, I'm not sure I buy it except in the case of conflicting services (nginx vs apache example) but this would be unusual and there are probably simpler ways to handle it.  In general system services aren't going to be the sole dependent of any other service that isn't specifically enabled anyway, so just stopping the first one and then starting the second one isn't really going to cause the issue you describe in your example (with `x`, `y` and `g` - can you give a real example with real services?).

To change to `single` user mode from a normally booted system just execute `dinitctl start single; dinitctl stop boot` - that already works just fine.

So overall I'm not currently convinced this would be a generally useful thing.

I also think the implementation solution proposed (with placeholder service) would be quite complex, in order to correctly handle all the edge cases, especially as placeholder services are already used in a very different context. (Sorry to be negative). However, there might be a simpler way. If you can devise a solution which is simple enough and correct in all scenarios then we can discuss it further, but I won't commit to accepting changes until that's proven.

To summarise:

- I don't really agree it's generally useful (I am open to being presented with evidence that shows otherwise)
- However, if you can implement it without adding too much complexity, then maybe I will accept it anyway
- But, you will run the risk that I'm not happy with the solution. So I recommend that you don't spend too much time on it. If it requires a lot of time, that indicates it is too complex anyway.
